qiita_pet/uimodules/prep_template_tab.py

```diff
@@ -35,13 +35,16 @@
 
 
 def _get_accessible_raw_data(user):
-    """Retrieves a tuple of raw_data_id and the last study title for that
+    """Retrieves a tuple of raw_data_id and one study title for that
     raw_data
     """
     d = {}
-    for sid in user.user_studies:
-        for rdid in Study(sid).raw_data():
-            d[int(rdid)] = Study(RawData(rdid).studies[-1]).title
+    accessible_studies = user.user_studies.union(user.shared_studies)
+    for sid in accessible_studies:
+        study = Study(sid)
+        study_title = study.title
+        for rdid in study.raw_data():
```

Comment:
syntactic, non-blocking, and purely for note: this could be done as `d.update({int(rdid): study_title for rdid in study.raw_data()})`

...out of curiosity, I benched these, and the insert is faster.

```python
19:48:26 (mcdonadt@Daniels-MBP-3):/Users/mcdonadt
> def insert():
    d = {}
    for i in xrange(100):
        for j in xrange(100):
            d[(i, j)] = i
   ...:

19:49:22 (mcdonadt@Daniels-MBP-3):/Users/mcdonadt
> def merge():
   ...:     d = {}
   ...:     for i in xrange(100):
   ...:         d.update({(i, j): i for j in xrange(100)})
   ...:

19:49:51 (mcdonadt@Daniels-MBP-3):/Users/mcdonadt
> %timeit insert()
1000 loops, best of 3: 1.49 ms per loop

19:50:02 (mcdonadt@Daniels-MBP-3):/Users/mcdonadt
> %timeit merge()
1000 loops, best of 3: 1.97 ms per loop
```